Lib/test/pickletester.py

```diff
@@ -2994,6 +2995,59 @@ def __reduce__(self):
 class BBB(object):
     pass
 
+
+class AbstractHookTests(unittest.TestCase):
```

Comment:
I agree, although it is not assumed that `_pickle` exists necessarily when the tests are ran. Therefore all the `CPickle*` test classes are created conditionally on the variable `has_c_implementation`, that is created in the `test_pickle` file, and not `pickletester`. I fear that creating directly the `CPicklerHookTests` is going to generate more complications and confusions than it will solve. I can leave a comment saying that the `global_hook` semantics only make sense for the the C implementation of `pickle` though.

Reply:
As discussed, I think the same hooks should be supported in Python and in C, even though the new hook is mostly useful with the C implementation.